mmv1/third_party/tgc_next/test/setup.go

```diff
@@ -45,49 +50,55 @@ type Resource struct {
 	Attributes map[string]struct{} `json:"attributes"`
 }
 
+const ymdFormat = "2006-01-02"
+
 var (
-	TestsMetadata = make(map[string]TgcMetadataPayload)
+	TestsMetadata = make([]NightlyRun, 30)
 	setupDone     = false
 )
 
-func ReadTestsDataFromGcs() (map[string]TgcMetadataPayload, error) {
+func ReadTestsDataFromGcs() ([]NightlyRun, error) {
 	if !setupDone {
 		bucketName := "cai_assets_metadata"
 		currentDate := time.Now()
+		ctx := context.Background()
+		client, err := storage.NewClient(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("storage.NewClient: %v", err)
+		}
+		defer client.Close()
 
-		for len(TestsMetadata) == 0 {
-			objectName := fmt.Sprintf("nightly_tests/%s/nightly_tests_meta.json", currentDate.Format("2006-01-02"))
-			log.Printf("Read object  %s from the bucket %s", objectName, bucketName)
-
-			ctx := context.Background()
-			client, err := storage.NewClient(ctx)
-			if err != nil {
-				return nil, fmt.Errorf("storage.NewClient: %v", err)
-			}
-			defer client.Close()
-
-			currentDate = currentDate.AddDate(0, 0, -1)
+		bucket := client.Bucket(bucketName)
 
-			rc, err := client.Bucket(bucketName).Object(objectName).NewReader(ctx)
+		var allErrs error
+		retriesRemaining := 30
+		for i := 0; i < len(TestsMetadata); i++ {
+			metadata, err := readTestsDataFromGCSForRun(ctx, currentDate, bucketName, bucket)
 			if err != nil {
-				if err == storage.ErrObjectNotExist {
-					log.Printf("Object '%s' in bucket '%s' does NOT exist.\n", objectName, bucketName)
-					continue
+				if allErrs == nil {
+					allErrs = fmt.Errorf("reading tests data from gcs: %v", err)
 				} else {
-					return nil, fmt.Errorf("Object(%q).NewReader: %v", objectName, err)
+					allErrs = fmt.Errorf("%v, %v", allErrs, err)
 				}
 			}
-			defer rc.Close()
-
-			data, err := io.ReadAll(rc)
-			if err != nil {
-				return nil, fmt.Errorf("io.ReadAll: %v", err)
+			if metadata == nil {
+				// Keep looking until we find a date with metadata.
+				i--
+				retriesRemaining--
```

Comment:
In what use cases  `retriesRemaining` is used? Is `i` enough to download the data for 30 days?